signalclient.go

```diff
@@ -81,26 +81,29 @@ func (c *SignalClient) Join(urlPrefix string, token string, params *ConnectParam
 	}
 
 	header := newHeaderWithToken(token)
-	conn, _, err := websocket.DefaultDialer.Dial(u.String(), header)
+	conn, hresp, err := websocket.DefaultDialer.Dial(u.String(), header)
 	if err != nil {
+		logger.Errorw("error establishing signal connection", err, "httpResponse", hresp)
```

Comment:
did you mean to log the entire response obj? or just the code?

I think when error is returned, response would might be nil

Reply:
I intended to log as much as the response to get more information. So, yeah, will have to dial that back once I have some data. Gorilla documentation says that if there is a hanshake failure, it can return an error with a HTTP response non-nil (https://github.com/gorilla/websocket/blob/af47554f343b4675b30172ac301638d350db34a5/client.go#L149). That's why I added it. I want to see if there is a clear indication of Dial error vs application error.

Speaking of that, let me add a log for the validate end point get error also if any to check what that returns.